### ISSUE TITLE
vtkDICOMParser cannot read DICOM files with non-ASCII characters in the filename

### ISSUE THREAD

**Issue by dmgroppe** [dgobbi/vtk-dicom#212]
A user received some DICOM files from clinical software and the filenames had a non-ASCII dot character in the name (see attached image).
<img width="426" alt="nonAsciiDot" src="https://user-images.githubusercontent.com/4405343/170775701-8384e438-8609-4069-8399-8b520fc8778c.png">

vtkDICOMParser could not read these files.

However, if we removed the non-ASCII dot, vtkDCIOMParser worked fine.

**Comment by dgobbi** [comment]
That's unexpected, the parser should be able to read non-ascii filenames.  Was this on Windows?  

Did you try reading it with the command-line tools (dicomdump, dicomtocsv, etc) or are you using vtkDICOMParser from your own C++ program?  Note that `SetFileName()` only takes ASCII or UTF-8, so if you give it any other encoding it will fail.

The DICOM command-line tools all use `wmain()` to get the command-line parameters as UTF-16, and then convert them to UTF-8 for use with the vtk-dicom API.

**Comment by dmgroppe** [comment]
Yes, this was on Windows. The error either occurred when running dicomfind.exe or dicomdump.exe.

The error message was "ERROR: In E:\Kitware\vtk-dicom\Source\vtkDICOMParser.cxx line 1871 vtkDICOMParser (02EDEDD8): ReadFile: Can't open the file"

**Comment by dgobbi** [comment]
I have not been able to reproduce this issue.  I've tried several languages and several tools (dicomfind.exe, dicomtocsv.exe, dicomdump.exe) and the special characters work fine for me, including a middle dot character:

```text
dicomdump "F:\DICOM\画像CD発行システム‎\山田‧太郎.dcm"
==== F:\DICOM\画像CD発行システム‎\山田‧太郎.dcm ====
(0002,0000) UL "FileMetaInformationGroupLength" : [188] (4 bytes)
(0002,0001) OB "FileMetaInformationVersion" : [...] (2 bytes)
(0002,0002) UI "MediaStorageSOPClassUID" : [1.2.840.10008.5.1.4.1.1.7] {Secondary Capture Image Storage} (26 bytes)
(0002,0003) UI "MediaStorageSOPInstanceUID" : [1.3.6.1.4.1.5962.1.1.0.1.1.1175775771.5702.0] (44 bytes)
(0002,0010) UI "TransferSyntaxUID" : [1.2.840.10008.1.2.1] {Explicit VR Little Endian} (20 bytes)
```
Can you send me the exact text of the problem filename?  A cut-and-paste from the console should work fine.  No screenshots, since it's the code for the characters and not their appearance that matters.

**Comment by dgobbi** [comment]
Closing because I've tried to reproduce this problem, but files with special characters work fine for me.  Will re-open if you can provide a .zip or .tar file with a DICOM file that demonstrates the problem.

**Comment by dgobbi** [comment]
Well, I think you mentioned before that you use vtkDICOMParser in C#?  Are you sure the filenames are converted to utf-8 before you call SetFileName()?

But that doesn't explain why you would see problems with dicomdump and dicomtocsv, since these are pure C++ programs and they access the filesystem with UTF-16.

**Comment by dgobbi** [comment]
A late follow-up on this issue.  For compressed DICOM files, with non-ASCII filenames, and only on Windows, the vtkDICOMReader had an incorrect hand-off of the filename to GDCM for GDCM 3.0.1 and later.  This has been fixed by f04f754ee in the latest release (0.8.16).
